internal/util/projutil/project_util.go

```diff
@@ -140,7 +152,52 @@ func getHomeDir() (string, error) {
 // under $GOPATH and returns the current directory's import path,
 // e.g: "github.com/example-inc/app-operator"
 func CheckAndGetProjectGoPkg() string {
-	gopath := MustSetGopath(MustGetGopath())
+	gopath := MustSetFirstGopath(MustGetGopath())
+	return parseGoPkg(gopath)
+}
+
+// GetGoPkg returns the current directory's import path by parsing it from
+// wd if this project's repository path is rooted under $GOPATH/src, or
+// from go.mod the project uses go modules to manage dependencies.
+//
+// Example: "github.com/example-inc/app-operator"
+func GetGoPkg() string {
+	// Default to reading from go.mod, as it should usually have the (correct)
+	// package path, and no further processing need be done on it if so.
+	if _, err := os.Stat(goModFile); err == nil {
+		b, err := ioutil.ReadFile(goModFile)
+		if err != nil {
+			log.Fatalf("Read go.mod: %v", err)
+		}
+		mf, err := modfile.Parse(goModFile, b, nil)
+		if err != nil {
+			log.Fatalf("Parse go.mod: %v", err)
+		}
+		if mf.Module != nil && mf.Module.Mod.Path != "" {
```

Comment:
If this evaluates to `false` it means there's a `go.mod` that was parsed successful that doesn't have a module declaration, right?
a) Is that even possible?
b) If so, would it be worth logging a fatal error and exiting or maybe just a logging warning that the module path is missing from go.mod so we're falling back to GOPATH-based detection.

Reply:
It is possible to have a `go.mod` without a module line, but only if we're in `$GOPATH/src`.